kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/TopicNameResult.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+
+import org.apache.kafka.common.Uuid;
+
+/**
+ * The result of an attempted topic id to topic name mapping. This is intended to enable different patterns for accessing the results.
+ */
+public interface TopicNameResult {
```

Comment:
I'd expose a single API - either TopicNameResult#topicNameMapping or TopicNameResult#topicNames, but not both.  I have a slight preference for the latter, but I think other feel different. Fine, let's go with that. 

If it turns out we need  TopicNameResult#topicNames, we can add it later.

Reply:
done I've removed `topicNames`